assig1/src/main/java/ro/utcn/sd/btn/assig1/persistence/memory/InMemoryRepositoryFactory.java

```diff
@@ -0,0 +1,34 @@
+package ro.utcn.sd.btn.assig1.persistence.memory;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import ro.utcn.sd.btn.assig1.persistence.api.QuestionRepository;
+import ro.utcn.sd.btn.assig1.persistence.api.RepositoryFactory;
+import ro.utcn.sd.btn.assig1.persistence.api.TagRepository;
+import ro.utcn.sd.btn.assig1.persistence.api.UserRepository;
+import ro.utcn.sd.btn.assig1.persistence.memory.InMemoryQuestionRepository;
+
+
+@Component
+@ConditionalOnProperty(name = "assig1.repository-type", havingValue = "MEMORY")
+public class InMemoryRepositoryFactory implements RepositoryFactory {
+    private final InMemoryQuestionRepository repository = new InMemoryQuestionRepository();
+    private final InMemoryUserRepository repository1 = new InMemoryUserRepository();
+    private final InMemoryTagRepository repository2 = new InMemoryTagRepository();
```

Comment:
Very bad naming! Exactly what we discussed at the lab... Please do not use "indexed" fields like this and give meaningful names instead.